qir/qat/Passes/TargetQisMappingPass/TransformationRulesPassConfiguration.cpp

```diff
@@ -0,0 +1,267 @@
+
+    // Experimental settings
+    config.addExperimentalParameter(delete_dead_code_, true, false, "delete-dead-code", "Deleted dead code.");
+    config.addExperimentalParameter(
+        clone_functions_, true, false, "clone-functions", "Clone functions to ensure correct qubit allocation.");
+    config.addExperimentalParameter(
+        transform_execution_path_only_, true, false, "transform-execution-path-only",
+        "Transform execution paths only.");
+
+    config.addExperimentalParameter(
+        max_recursion_, max_recursion_, uint64_t(1), "max-recursion",
+        "Defines the maximum recursion when unrolling the execution path");
+
+    config.addExperimentalParameter(
+        assume_no_exceptions_, false, false, "assume-no-except",
+        "Assumes that no exception will occur during runtime.");
+
+    config.addParameter(reuse_qubits_, false, "reuse-qubits", "Use to define whether or not to reuse qubits.");
+    config.addParameter(reuse_results_, false, "reuse-results", "Use to define whether or not to reuse results.");
+
+    // Rules
+    config.addParameter(
+        optimize_result_one_, true, "optimize-result-one",
+        "Maps branching based on quantum measurements compared to one to base adaptor "
+        "type measurement.");
+    config.addParameter(
+        optimize_result_zero_, true, "optimize-result-zero",
+        "Maps branching based on quantum measurements compared to zero to base adaptor "
+        "type measurement.");
+    config.addParameter(
+        optimize_result_comparison_, true, "optimize-result-comparison",
+        "Maps branching based on quantum measurements compared to each other to base adaptor "
+        "type measurement.");
+    config.addParameter(
+        use_static_qubit_array_allocation_, "use-static-qubit-array-allocation",
+        "Maps allocation of qubit arrays to static array allocation.");
+    config.addParameter(
+        use_static_qubit_allocation_, "use-static-qubit-allocation", "Maps qubit allocation to static allocation.");
+    config.addParameter(
+        use_static_result_allocation_, "use-static-result-allocation", "Maps result allocation to static allocation.");
+
+    config.addParameter(
+        disable_reference_counting_, true, "disable-reference-counting",
+        "Disables reference counting by instruction removal.");
+    config.addParameter(
+        disable_alias_counting_, true, "disable-alias-counting", "Disables alias counting by instruction removal.");
+    config.addParameter(
+        disable_string_support_, true, "disable-string-support", "Disables string support by instruction removal.");
+    config.addParameter(
+        disable_record_output_support_, true, "disable-record-output-support",
+        "Disables record output support by instruction removal.");
+
+    // Ready settings
+
+    config.addParameter(entry_point_attr_, "entry-point-attr", "Specifies the attribute indicating the entry point.");
```

Comment:
How did you deal with the name change for the attributes containing the required number of qubits and results for the entry point? These also changed with the latest spec compared to the original implementation. Is there currently no support yet for generating and validating ir that is compliant with the official spec under development?

Reply:
It is not within the scope of this PR to deal with the name change. As previously pointed out, someone would need to write a pass/adaptor that would perform said change. As I recall it, we agreed that this would be a good starting task for someone picking up QAT.